### PR TITLE
Update SDK to 6.0.100-rc.1.21410.3

### PR DESCRIPTION
**Pull request by pranavkm**

### CODE REVIEW

**Comment by pranavkm** [comment]
Looks like this is still running in to https://github.com/dotnet/roslyn/issues/55461. We'll have to wait for a new installer before the SDK can be updated.

**Comment by sebastienros** [comment]
This test fails:

https://github.com/dotnet/aspnetcore/blob/main/src/Razor/Microsoft.AspNetCore.Razor.Language/test/IntegrationTests/ComponentDirectiveIntegrationTest.cs#L95

with

```
 Microsoft.AspNetCore.Razor.Language.IntegrationTests.ComponentDirectiveIntegrationTest.SupportsInjectDirective
Microsoft.AspNetCore.Razor.Language.IntegrationTests.RazorIntegrationTestBase+CompilationFailedException : Compilation failed: \n/dir/subdir/Test/TestComponent.cshtml(3,120): error CS0103: The name 'MyService1' does not exist in the current context\n/dir/subdir/Test/TestComponent.cshtml(3,136): error CS0103: The name 'MyService2' does not exist in the current context\n(10,5): hidden CS8019: Unnecessary using directive.\n(7,5): hidden CS8019: Unnecessary using directive.\n(9,5): hidden CS8019: Unnecessary using directive.\n(8,5): hidden CS8019: Unnecessary using directive.\n(6,5): hidden CS8019: Unnecessary using directive.\n\n\nFile :\n// <auto-generated/>\n#pragma warning disable 1591\nnamespace Test\n{\n    #line hidden\n    using System;\n    using System.Collections.Generic;\n    using System.Linq;\n    using System.Threading.Tasks;\n    using Microsoft.AspNetCore.Components;\n    public partial class TestComponent : Microsoft.AspNetCore.Components.ComponentBase\n    {\n        #pragma warning disable 1998\n        protected override void BuildRenderTree(Microsoft.AspNetCore.Components.Rendering.RenderTreeBuilder __builder)\n        {\n            __builder.AddMarkupContent(0, \"MyService1\\nMicrosoft.AspNetCore.Razor.Language.IntegrationTests.ComponentDirectiveIntegrationTest.IMyService1@inject  MyService2\\nMicrosoft.AspNetCore.Razor.Language.IntegrationTests.ComponentDirectiveIntegrationTest.IMyService2@inject Hello from \");\n#nullable restore\n#line (3,119)-(3,129) 24 \"/dir/subdir/Test/TestComponent.cshtml\"\n__builder.AddContent(1, MyService1);\n\n#line default\n#line hidden\n#nullable disable\n            __builder.AddContent(2, \" and \");\n#nullable restore\n#line (3,135)-(3,145) 24 \"/dir/subdir/Test/TestComponent.cshtml\"\n__builder.AddContent(3, MyService2);\n\n#line default\n#line hidden\n#nullable disable\n        }\n        #pragma warning restore 1998\n    }\n}\n#pragma warning restore 1591\n\n
   at Microsoft.AspNetCore.Razor.Language.IntegrationTests.RazorIntegrationTestBase.CompileToAssembly(CompileToCSharpResult cSharpResult, Boolean throwOnFailure) in /_/src/Razor/test/Microsoft.AspNetCore.Razor.Test.Common/Language/IntegrationTests/RazorIntegrationTestBase.cs:line 305
   at Microsoft.AspNetCore.Razor.Language.IntegrationTests.RazorIntegrationTestBase.CompileToAssembly(String cshtmlRelativePath, String cshtmlContent) in /_/src/Razor/test/Microsoft.AspNetCore.Razor.Test.Common/Language/IntegrationTests/RazorIntegrationTestBase.cs:line 281
   at Microsoft.AspNetCore.Razor.Language.IntegrationTests.RazorIntegrationTestBase.CompileToComponent(String cshtmlSource) in /_/src/Razor/test/Microsoft.AspNetCore.Razor.Test.Common/Language/IntegrationTests/RazorIntegrationTestBase.cs:line 331
   at Microsoft.AspNetCore.Razor.Language.IntegrationTests.ComponentDirectiveIntegrationTest.SupportsInjectDirective() in /_/src/Razor/Microsoft.AspNetCore.Razor.Language/test/IntegrationTests/ComponentDirectiveIntegrationTest.cs:line 98
```

There are other tests failing but I believe they have the same root cause. @pranavkm can you take a quick look? If you give me some guidance I can apply the changes for you.

**Comment by wtgodbe** [comment]
@pranavkm what are the non-SDK changes for?

**Comment by pranavkm** [comment]
> @pranavkm what are the non-SDK changes for?

Trimmer changes and fixing logging analyzer warnings.

**Comment by wtgodbe** [comment]
Well now we're very broken:

> /Users/runner/work/1/s/src/Components/Components/src/NavigationManagerExtensions.cs(47,38): error CS0246: The type or namespace name 'DateTime' could not be found (are you missing a using directive or an assembly reference?) [/Users/runner/work/1/s/src/Components/Components/src/Microsoft.AspNetCore.Components.csproj]

Did these recently move? Looking

**Comment by pranavkm** [comment]
These files probably expect the implicit usings which were removed. For now, adding the usings back or adding `ImplicitUsings=enable` to the root `Directory.Build.props` would be the way to go.

**Comment by wtgodbe** [comment]
Too much onion peeling, I'm just gonna enable implicit usings.

File: Directory.Build.props [hunk 1]
```diff
@@ -50,6 +50,11 @@
     <DisableImplicitComponentsAnalyzers>true</DisableImplicitComponentsAnalyzers>
   </PropertyGroup>
 
+  <!-- Enable implicit usings -->
+  <PropertyGroup>
+    <ImplicitUsings>enable</ImplicitUsings>
```
Comment (pranavkm) [inline Directory.Build.props:55]: One maybe thing to do would be to turn it off if there's more than one TFM. It'll avoid fighting VS tooling. 

```suggestion
    <ImplicitUsings Condition="''$(TargetFrameworks)' == ''">enable</ImplicitUsings>
```
I guess that would require specifying this in the Directory.Build.targets

**Review by wtgodbe — APPROVED** [review]

**Pull request merged**